### DISCUSSION TITLE
Brew update on gh actions slow due to reinstall of gcc@8 from src

### DISCUSSION THREAD

**Discussion by johnny-bit** [Homebrew/discussions#183] [Everyday usage] [answered]
I'm using GH actions provided `macos-latest` macos for darktable CI runs and inside those, I'm using brew to install dependencies.

Roughly 2 days ago it looks as if macos build slowed down to a crawl, due to:

```
Sun, 29 Nov 2020 16:14:16 GMT ==> Checking for dependents of upgraded formulae...
Sun, 29 Nov 2020 16:14:16 GMT ==> Reinstalling 1 broken dependent from source:
Sun, 29 Nov 2020 16:14:16 GMT gcc@8
Sun, 29 Nov 2020 16:14:17 GMT ==> Downloading https://ftp.gnu.org/gnu/gcc/gcc-8.4.0/gcc-8.4.0.tar.xz
Sun, 29 Nov 2020 16:14:18 GMT ==> Reinstalling gcc@8 
Sun, 29 Nov 2020 16:15:55 GMT ==> ../configure --build=x86_64-apple-darwin19 --prefix=/usr/local/Cellar/gcc@8/8.4.0_1 --libdir=/usr/local/Cellar/gcc@8/8.4.0_1/lib/gcc/8 --disable-nls --enable-checking=release --enable-languages=c,c++,objc,obj-c++,fortran --program-suffix=-8 --with-gmp=/usr/local/opt/gmp --with-mpfr=/usr/local/opt/mpfr --with-mpc=/usr/local/opt/libmpc --with-isl=/usr/local/opt/isl --with-system-zlib --with-pkgversion=Homebrew GCC 8.4.0_1 --with-bugurl=https://github.com/Homebrew/homebrew-core/issues --disable-multilib --with-native-system-header-dir=/usr/include --with-sysroot=/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk SED=/usr/bin/sed
Sun, 29 Nov 2020 16:16:07 GMT ==> make BOOT_LDFLAGS=-Wl,-headerpad_max_install_names
Sun, 29 Nov 2020 16:54:30 GMT ==> make install
Sun, 29 Nov 2020 16:55:04 GMT 🍺  /usr/local/Cellar/gcc@8/8.4.0_1: 1,415 files, 275MB, built in 40 minutes 45 seconds
Sun, 29 Nov 2020 16:55:04 GMT ==> Caveats
```

Brewfile is here: https://github.com/darktable-org/darktable/blob/master/.ci/Brewfile
Github ci run workflow is here: https://github.com/darktable-org/darktable/blob/master/.github/workflows/ci.yml
With relevant part (brew setup) is here:
https://github.com/darktable-org/darktable/blob/ce13a3c134ef6b487133aebae3ca86acd3d72b95/.github/workflows/ci.yml#L323-L328

Additionally - I had a look at https://github.com/Homebrew/actions/tree/master/setup-homebrew but I don't know if I can use it in my (darktable's) situation for caching dependencies (which would IMO greatly speedup whole process :) )

**Comment by SMillerDev** [comment]
I suspect that there is more output before this error. Do you have a link to the run that failed like this?

---

> Additionally - I had a look at https://github.com/Homebrew/actions/tree/master/setup-homebrew but I don't know if I can use it in my (darktable's) situation for caching dependencies (which would IMO greatly speedup whole process :) )

That's just useful to cache the brew command dependencies.

**Reply by johnny-bit** [reply]
The run didn't fail... just takes annoyingly long time ;)
Here's the sample recent build: https://github.com/darktable-org/darktable/runs/1472467217?check_suite_focus=true#step:3:1 (step "Install base dependencies")

**Comment by SMillerDev** [comment, marked as answer]
It could very well be caused by a change on the public runners that trigger: https://github.com/Homebrew/homebrew-core/blob/master/Formula/gcc@8.rb#L21-L26

**Reply by johnny-bit** [reply]
OK... Is there something we can do to not rebuild gcc8 every time? :)

**Reply by SMillerDev** [reply]
Use your own runners or request that GitHub actions runners provide a CLT install.

**Reply by maxim-lobanov** [reply]
GitHub Hosted runners already contains CLT installed.
If Homebrew find "broken" packages, it always rebuilds them from source. Looks like it is by design. See related code: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/upgrade.rb#L240

**Reply by henkrehorst** [reply]
I have the same issue with Azure Pipelines, gcc takes longer than an hour to compile and Azure Pipelines has a maximum build time of 1 hour. I am also using the most recent version of xcode. What is the best solution to solve this problem.

**Reply by SMillerDev** [reply]
I can't help much beyond "make sure GCC doesn't need to compile". From what I can see from the formula that would mean it needs the CLT installed.

**Reply by carlocab** [reply]
@maxim-lobanov CLT doesn't seem to be listed here? https://github.com/actions/virtual-environments/blob/macOS-10.15/20201115.1/images/macos/macos-10.15-Readme.md

Or have I missed it?

**Reply by SMillerDev** [reply]
Doesn't seem that way to me either.

**Reply by carlocab** [reply]
One way to check what brew thinks about whether CLT is installed or not: do `brew config` on your next CI run.

**Reply by maxim-lobanov** [reply]
CLT exists on the Hosted images and brew sees it.
`gcc@8` formula is broken on current image and will be fixed with next image deployment.
If formula is broken, brew always compile it from source code on re-installing even if CTL is installed: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/upgrade.rb#L240
